File: typescript-examples/browser-sdk-showcase/api/ai/is-page-loaded.ts
// https://docs.intunedhq.com/automation-sdks/intuned-sdk/typescript/ai/functions/isPageLoaded
import { BrowserContext, Page } from "playwright";
import { isPageLoaded } from "@intuned/browser/ai";

interface Params {
  // No params needed
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  // Wait for page to finish loading
  await page.goto("https://sandbox.intuned.dev/");
  
  const pageLoaded = await isPageLoaded({ page });
  if (pageLoaded) {
    // Continue with scraping or interactions
    console.log("Page is loaded");
  } else {
    // Wait longer or retry
    await page.waitForTimeout(5000);
  }
}

File: python-examples/browser-sdk-showcase/api/ai/is-page-loaded.py
# https://docs.intunedhq.com/automation-sdks/intuned-sdk/python/ai/functions/is_page_loaded
from typing import TypedDict

from intuned_browser.ai import is_page_loaded
from playwright.async_api import Page


class Params(TypedDict):
    pass


async def automation(page: Page, params: Params, **_kwargs):
    # Wait for page to finish loading
    await page.goto("https://sandbox.intuned.dev/")
    page_loaded = await is_page_loaded(page)
    if page_loaded:
        # Continue with scraping or interactions
        print("Page is loaded")
    else:
        # Wait longer or retry
        await page.wait_for_timeout(5000)
